packages/react/src/domain/handles.ts

```diff
@@ -58,6 +61,13 @@ function getPossibleHandleTypes(
     handleTypes.add("phone_number");
   }
 
+  if (
+    FACTORS_WITH_USERNAME.includes(factor.method) &&
+    isHandleTypeAllowed(factor, "username")
```

Comment:
I feel like we could combine these two in one fn for simplicity

Reply:
I understand the concern, right now I don't see an obvious and easy way of making this more concise so I'd rather skip this in the scope of this PR. Happy to go back and refactor once there's a clear idea on how to do it exactly